src/test/ui/traits/traits-assoc-type-in-supertrait-bad.stderr

```diff
@@ -1,8 +1,8 @@
 error[E0271]: type mismatch resolving `<std::vec::IntoIter<i32> as std::iter::Iterator>::Item == u32`
-  --> $DIR/traits-assoc-type-in-supertrait-bad.rs:11:6
+  --> $DIR/traits-assoc-type-in-supertrait-bad.rs:12:16
    |
-LL | impl Foo for IntoIter<i32> {
-   |      ^^^ expected `i32`, found `u32`
+LL |     type Key = u32;
+   |                ^^^ expected `i32`, found `u32`
```

Comment:
I'm looking at keeping enough information around for obligations born out of projections, but this might take a while. At the same time, these are the ones that will be the most useful to fix.